schema_utils_test.py

```diff
@@ -581,3 +653,42 @@ def test_notification_email_list_validator(self):
         invalid_vals = [[u'admin@oppia'], big_email_list,
                         [u'admin@oppia.commmm'], [u'a@.com']]
         self.check_normalization(schema, mappings, invalid_vals)
+
+    def test_normalize_spaces(self):
+        """Test static method normalize_spaces; should collapse multiple
+        spaces.
+        """
+        normalize_spaces = schema_utils.Normalizers.get('normalize_spaces')
+        self.assertEqual('dog cat', normalize_spaces('dog     cat'))
+
+    def test_normalizer_get(self):
+        """Tests the class method get of Normalizers, should return the
+        normalizer method corresponding to the given normalizer id.
+        """
+
+        # Will test .get on the normalize_spaces method.
+        normalize_spaces = schema_utils.Normalizers.get('normalize_spaces')
+        self.assertEqual('normalize_spaces', normalize_spaces.__name__)
+
+    def test_normalizer_get_raises_exception_for_invalid_id(self):
+        """Tests if class method get of Normalizers raises exception when given
+        an invalid normalizer id.
+        """
+        with self.assertRaises(Exception):
+            schema_utils.Normalizers.get('some invalid normalizer method name')
+
+    def test_sanitize_url_functionality(self):
+        """Tests if static method sanitize_url of Normalizers correctly
+        sanitizes a URL when given its string representation.
+        """
+        sanitize_url_fn = schema_utils.Normalizers.get('sanitize_url')
+        (self.assertEqual('https://www.oppia.org/splash/',
+                          sanitize_url_fn('https://www.oppia.org/splash/')))
+
+    def test_sanitize_url_raises_error_for_invalid_url(self):
+        """Tests if static method sanitize_url of Normalizers raises error
+        for invalid URL.
+        """
+        sanitize_url_fn = schema_utils.Normalizers.get('sanitize_url')
+        with self.assertRaises(AssertionError):
+            sanitize_url_fn('oppia.org') # missing https:// throws assert error
```

Comment:
I think you can combine this and the above test.

Check all other cases you can think of: http://, other initial protocols, etc.

Reply:
addressed in next commit/pr